openstudiocore/src/model/SiteGroundTemperatureDeep.cpp

```diff
@@ -0,0 +1,653 @@
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool SiteGroundTemperatureDeep_Impl::isAugustDeepGroundTemperatureDefaulted() const {
+    return isEmpty(OS_Site_GroundTemperature_DeepFields::AugustDeepGroundTemperature);
+  }
+
+  double SiteGroundTemperatureDeep_Impl::septemberDeepGroundTemperature() const {
+    boost::optional<double> value = getDouble(OS_Site_GroundTemperature_DeepFields::SeptemberDeepGroundTemperature,true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool SiteGroundTemperatureDeep_Impl::isSeptemberDeepGroundTemperatureDefaulted() const {
+    return isEmpty(OS_Site_GroundTemperature_DeepFields::SeptemberDeepGroundTemperature);
+  }
+
+  double SiteGroundTemperatureDeep_Impl::octoberDeepGroundTemperature() const {
+    boost::optional<double> value = getDouble(OS_Site_GroundTemperature_DeepFields::OctoberDeepGroundTemperature,true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool SiteGroundTemperatureDeep_Impl::isOctoberDeepGroundTemperatureDefaulted() const {
+    return isEmpty(OS_Site_GroundTemperature_DeepFields::OctoberDeepGroundTemperature);
+  }
+
+  double SiteGroundTemperatureDeep_Impl::novemberDeepGroundTemperature() const {
+    boost::optional<double> value = getDouble(OS_Site_GroundTemperature_DeepFields::NovemberDeepGroundTemperature,true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool SiteGroundTemperatureDeep_Impl::isNovemberDeepGroundTemperatureDefaulted() const {
+    return isEmpty(OS_Site_GroundTemperature_DeepFields::NovemberDeepGroundTemperature);
+  }
+
+  double SiteGroundTemperatureDeep_Impl::decemberDeepGroundTemperature() const {
+    boost::optional<double> value = getDouble(OS_Site_GroundTemperature_DeepFields::DecemberDeepGroundTemperature,true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool SiteGroundTemperatureDeep_Impl::isDecemberDeepGroundTemperatureDefaulted() const {
+    return isEmpty(OS_Site_GroundTemperature_DeepFields::DecemberDeepGroundTemperature);
+  }
+
+  double SiteGroundTemperatureDeep_Impl::getTemperatureByMonth(int month) const {
+    double retval;
+    switch(month) {
+      case 1 : retval = januaryDeepGroundTemperature(); break;
+      case 2 : retval = februaryDeepGroundTemperature(); break;
+      case 3 : retval = marchDeepGroundTemperature(); break;
+      case 4 : retval = aprilDeepGroundTemperature(); break;
+      case 5 : retval = mayDeepGroundTemperature(); break;
+      case 6 : retval = juneDeepGroundTemperature(); break;
+      case 7 : retval = julyDeepGroundTemperature(); break;
+      case 8 : retval = augustDeepGroundTemperature(); break;
+      case 9 : retval = septemberDeepGroundTemperature(); break;
+      case 10 : retval = octoberDeepGroundTemperature(); break;
+      case 11 : retval = novemberDeepGroundTemperature(); break;
+      case 12 : retval = decemberDeepGroundTemperature(); break;
```

Comment:
Yeah, the overload can just call the version that takes an int.  You can get the int value using [enum.value()](https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/utilities/core/EnumBase.hpp#L90)